system_tests/tests/pearlpc.py

```diff
@@ -236,21 +236,13 @@ def test_WHEN_auto_mode_is_set_THEN_auto_mode_can_be_read_back(self):
         self.lewis.backdoor_set_on_device("am_mode", 1)
         self.ca.assert_that_pv_is("INPUTS:AUTO", "Active")
 
-    @contextlib.contextmanager
-    def _disconnect_device(self):
-        self.lewis.backdoor_set_on_device("connected", False)
-        try:
-            yield
-        finally:
-            self.lewis.backdoor_set_on_device("connected", True)
-
     @parameterized.expand(parameterized_list(INPUT_PVS))
     def test_WHEN_device_disconnected_THEN_inputs_go_into_alarm(self, _, pv):
         self.ca.assert_that_pv_alarm_is(pv, self.ca.Alarms.NONE)
 
-        with self._disconnect_device():
+        with self.lewis.backdoor_simulate_disconnected_device():
             self.ca.assert_that_pv_alarm_is(pv, self.ca.Alarms.INVALID)
-
+        # Assert alarms clear on reconnection
         self.ca.assert_that_pv_alarm_is(pv, self.ca.Alarms.NONE)
```

Comment:
How do you feel about placing this into a separate private method or new test entirely?